### ISSUE TITLE
Improving search efficiency and speed

### ISSUE THREAD

**Issue by skenow** [ImpressCMS/impresscms#1368]
As reported in the forums by MadFish - the current search strategy carries a lot of overhead, especially when there are a lot of results. He provided a proof of concept for reducing the query time significantly in this post - https://www.impresscms.org/modules/iforum/viewtopic.php?post_id=47912#forumpost47912

**Comment by skenow** [comment]
Maybe this isn't so much a core thing as it is a module thing. Where should it go?

**Comment by MekDrop** [comment]
From my understanding - both. This place needs refactoring into something that modules gives parameters where to search and core uses them. Right now module asks all modules for search results and then merges. And that uses a lot of resources.

I don't see how we could make something good for 1.5.x. Only maybe for 2.0.x.

**Comment by fiammybe** [comment]
I agree that the focus for this should be on 2.x. 2.0 if it won't delay that version too much, otherwise a follow up release. First thing is now to get 2.0 stable on shared hosting.
